examples/log.rs

```diff
@@ -1,99 +1,114 @@
+    Topic2(U256, U256),
+    /// Event with three topics
+    Topic3(U256, U256, U256),
+    /// Event with four topics 
+    Topic4(U256, U256, U256, U256),
 }
 
-#[zink::external]
-pub fn log2() {
-    Ping.log2(b"pong", b"ping");
+pub mod event_tests {
+    use super::*;
+
+    /// Test log0 
+    #[zink::external]
+    pub fn test_log0() {
+        unsafe { zink::ffi::evm::log0(b"MyEvent") }
+    }
+
+    /// Test log1 
+    #[zink::external]
+    pub fn test_log1(value: U256) {
+        unsafe {
+            let topic = value.to_bytes32();
+            zink::ffi::evm::log1(b"MyEvent", topic)
+        }
+    }
+
+    /// Test log2 
+    #[zink::external]
+    pub fn test_log2(value1: U256, value2: U256) {
+        unsafe {
+            let topic1 = value1.to_bytes32();
+            let topic2 = value2.to_bytes32();
+            zink::ffi::evm::log2(b"MyEvent", topic1, topic2)
+        }
+    }
+
+    /// Test log3 
+    #[zink::external]
+    pub fn test_log3(value1: U256, value2: U256, value3: U256) {
+        unsafe {
+            let topic1 = value1.to_bytes32();
+            let topic2 = value2.to_bytes32();
+            let topic3 = value3.to_bytes32(); 
+            zink::ffi::evm::log3(b"MyEvent", topic1, topic2, topic3)
+        }
+    }
+
+    /// Test log4 
+    #[zink::external]
+    pub fn test_log4(value1: U256, value2: U256, value3: U256, value4: U256) {
+        unsafe {
+            let topic1 = value1.to_bytes32();
+            let topic2 = value2.to_bytes32();
+            let topic3 = value3.to_bytes32();
+            let topic4 = value4.to_bytes32();
+            zink::ffi::evm::log4(b"MyEvent", topic1, topic2, topic3, topic4)
+        }
+    }
+
+    /// Test multiple event logs in one transaction
+    #[zink::external]
+    pub fn test_multiple_logs(
+        value1: U256,
+        value2: U256,
+        value3: U256,
+        value4: U256,
+    ) -> Result<(), ()> {
+        test_log0();
+        test_log1(value1);
+        test_log2(value1, value2);
+        test_log3(value1, value2, value3);
+        test_log4(value1, value2, value3, value4);
+        Ok(())
+    }
 }
 
-#[zink::external]
-pub fn log3() {
-    Ping.log3(b"pong", b"ping", b"pong");
-}
-
-#[zink::external]
-pub fn log4() {
-    Ping.log4(b"pong", b"ping", b"pong", b"pong");
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_events() {
+        let value1 = U256::from(U256::empty());
+        let value2 = U256::from(U256::empty());
+        let value3 = U256::from(U256::empty());
+        let value4 = U256::from(U256::empty());
+
+        // Test each log function
+        event_tests::test_log0();
+        event_tests::test_log1(value1);
+        event_tests::test_log2(value1, value2);
+        event_tests::test_log3(value1, value2, value3);
+        event_tests::test_log4(value1, value2, value3, value4);
+
+        // Test multiple logs
+        event_tests::test_multiple_logs(value1, value2, value3, value4).unwrap();
```

Comment:
should compile the contract and then call these interfaces to complete the tests

Reply:
@clearloop how do we assert the output, logs don't return yet

Reply:
for example here:

https://github.com/zink-lang/zink/blob/main/examples/log.rs#L44-L50

1. we deploy the contract to our testing EVM
2. we call the method, in the returned `info`, there is a log field, see https://docs.zink-lang.org/rustdocs/zint/struct.Info.html#structfield.logs